Apps/AdminWebClient/src/Server/Controllers/CsvExportController.cs

```diff
@@ -0,0 +1,114 @@
+//  Copyright © 2019 Province of British Columbia
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+// -------------------------------------------------------------------------
+namespace HealthGateway.Admin.Controllers
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.IO.Compression;
+    using System.Net;
+    using System.Net.Http;
+    using System.Net.Http.Headers;
+    using System.Threading.Tasks;
+    using HealthGateway.Admin.Services;
+    using HealthGateway.Common.Models;
+    using HealthGateway.Common.Utils;
+    using HealthGateway.Database.Models;
+    using Microsoft.AspNetCore.Authorization;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.AspNetCore.Mvc;
+
+    /// <summary>
+    /// Web API to handle user email interactions.
+    /// </summary>
+    [ApiController]
+    [ApiVersion("1.0")]
+    [Route("v{version:apiVersion}/api/[controller]")]
+    [Produces("application/json")]
+    [Authorize(Roles = "AdminUser")]
+    public class CsvExportController
+    {
+        private readonly ICsvExportService dataExportService;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CsvExportController"/> class.
+        /// </summary>
+        /// <param name="dataExportService">The injected data export service.</param>
+        public CsvExportController(ICsvExportService dataExportService)
+        {
+            this.dataExportService = dataExportService;
+        }
+
+        /// <summary>
+        /// Retrieves a list of User Profiles created inclusively between UTC dates if provided.
+        /// </summary>
+        /// <param name="startDate">The optional start date for the data.</param>
+        /// <param name="endDate">The optional end date for the data.</param>
+        /// <returns>A CSV of the raw data. email.</returns>
+        /// <response code="200">Returns the list of beta requests.</response>
+        /// <response code="401">the client must authenticate itself to get the requested response.</response>
+        /// <response code="403">The client does not have access rights to the content; that is, it is unauthorized, so the server is refusing to give the requested resource. Unlike 401, the client's identity is known to the server.</response>
+        [HttpGet]
+        [Produces("text/csv")]
+        public HttpResponseMessage GetUserProfiles(DateTime? startDate = null, DateTime? endDate = null)
```

Comment:
suggestions: can we update these inputs as follow?

- startDate to createdOnOrAfter
- endDate to createdOnOrBefore 
If agree, can we rename them all for this PR?